File: src/main/java/com/projectswg/forwarder/resources/networking/packets/KeepAlive.java
package com.projectswg.forwarder.resources.networking.packets;

import java.nio.ByteBuffer;

public class KeepAlive extends Packet {
	
	public KeepAlive() {
		
	}
	
	public KeepAlive(ByteBuffer data) {
		decode(data);
	}
	
	public void decode(ByteBuffer data) {
		data.position(2);
	}
	
	public ByteBuffer encode() {
		ByteBuffer data = ByteBuffer.allocate(2);
		addNetShort(data, 0x06);
		return data;
	}
	
	@Override
	public String toString() {
		return String.format("KeepAlive[]");
	}
	
}

File: src/main/java/com/projectswg/forwarder/resources/networking/packets/KeepAlive.kt
package com.projectswg.forwarder.resources.networking.packets

import com.projectswg.common.network.NetBuffer


class KeepAlive : Packet {
	
	constructor()
	constructor(data: NetBuffer) {
		decode(data)
	}
	
	override fun decode(data: NetBuffer) {
		data.position(2)
	}
	
	override fun encode(): NetBuffer {
		val data = NetBuffer.allocate(2)
		data.addNetShort(0x06)
		return data
	}
	
	override fun toString(): String {
		return "KeepAlive[]"
	}
	
}
